simpleclient_servlet/src/main/java/io/prometheus/client/filter/MetricsFilter.java

```diff
@@ -175,5 +184,6 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 
     @Override
     public void destroy() {
+      collectorRegistry.unregister(histogram);
```

Comment:
This shouldn't be removed, the values should continue to be exposed until the process terminates.

Reply:
I second that this should happen for metricsFilter: in our use-case a web application which uses MetricsFilter can not be hot-redeployed to the running tomcat due to the metric being already registered. With this fix it would work: on the filter destroy (happening on redeploy) the metrics would be reset. This in some cases is not ideal indeed, but the current way it breaks the whole idea of "hot-deployable" war files to a running container.